src/core/thread/mle_router.cpp

```diff
@@ -1859,7 +1857,8 @@ void MleRouter::HandleStateUpdateTimer(void)
                 // upgrade to Router
                 BecomeRouter(ThreadStatusTlv::kTooFewRouters);
             }
-            else if (!mAdvertiseTimer.IsRunning())
+
+            if (!mAdvertiseTimer.IsRunning())
             {
                 SendAdvertisement();
```

Comment:
Should we suppress this `SendAdvertisement()` if there is ongoing address solicit?
Suppose one parent is trying to attach to the same partition, there are chances that it sends out one REED's MLE Advertisement before receiving Address Solicit Response if the random `mRouterSelectionJitter` has a small value, thus may causing its children to re-attach, which should not be expected. Thoughts?